components/user-service/src/main/java/edu/jhuapl/dorset/ldapuserservice/LdapUserService.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2017 The Johns Hopkins University Applied Physics Laboratory LLC
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.jhuapl.dorset.ldapuserservice;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.typesafe.config.Config;
+
+import edu.jhuapl.dorset.users.User;
+import edu.jhuapl.dorset.users.UserException;
+import edu.jhuapl.dorset.users.UserService;
+
+public class LdapUserService implements UserService {
+    private static final Logger logger = LoggerFactory.getLogger(LdapUserService.class);
+
+    private Map<String, User> users;
+
+    private String ldapServer;
+    private String ldapSearchBase;
+    private String ldapPassword;
+    private String ldapBindDn;
+    private String ldapFilterAttribute;
+
+    private String[] userAttributes;
+    private DirContext ctx;
+    
+    /**
+     * 
+     * LdapUserService
+     * 
+     */
+    public LdapUserService(Config config) {
+        this.users = new HashMap<String, User>();
+        
+        Config conf = config;
+
+        this.ldapServer = conf.getString("ldap-context.ldapServer");
```

Comment:
I'd lean towards having the user pass in a config object without the namespace. Typesafe supports getting a new config object based on a namespace. This gives the user flexibility in where they store the config information.